Add light platform for Yeelight Sunflower lights

lindsaymarkward opened the pull request:
**Description:**
Add new platform for the light component for Yeelight Sunflower bulbs (https://www.yeelight.com/en_US/product/yeelight-sunflower).

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#2006

**Example entry for `configuration.yaml` (if applicable):**
```yaml
light:
  - platform: yeelightsunflower
    host: 192.168.1.59
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [x] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [x] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [x] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [x] New files were added to `.coveragerc`.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L16
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L51

lindsaymarkward committed:
Add light platform for Yeelight Sunflower lights

tdickman committed:
Fix abreviation for miles in darksky sensor

happyleavesaoc committed:
Limitless light: bump version; fix conf (#6006)

andrey-git commented:
Could write tests for this platform?
You can mock out the `yeelightsunflower` lib.

nodinosaur committed:
Added a config flag[Boolean] to declare if SenseHAT is attached  (#5883)

* Added a config flag[Boolean] to declare if SenseHAT is attached to RaspberryPi

* Hound found a line too long violation - fix

pvizeli commented on homeassistant/components/light/yeelightsunflower.py:
```diff
@@ -0,0 +1,124 @@
+    @property
+    def unique_id(self):
+        """Return the ID of this light."""
+        return "{}.{}".format(self.__class__, self._light.zid)
```
Remove this. Is not needed if it is the same as `name`

lindsaymarkward replied:
OK. They will be different, but I was just filling in all of the methods I saw in the parent that made sense. If name covers it, then I'll take out unique_id.

aequitas committed:
Reconnect robustness, expose connection state. (#5869)

* Reconnect robustness, expose connection state.

- Expose connection status as rflink.connection_status state.
- Handle alternative timeout scenario.
- Explicitly set a timeout for connections.
- Error when trying to send commands if disconnected.
- Do not block component setup on gateway connection.

* Don't use coroutine where none is needed.

* Test disconnected behaviour.

* Use proper conventions for task creation.

* Possibly fix test race condition?

* Update hass import style

postlund committed:
Add fake support for turn on/off for Apple TV (#5962)

* Add fake support for turn on/off for Apple TV

When the device is "turned off", no requests are sent to the device.
When the setting "start_off" is set to true, the device starts in off
state.

* Fix async comments

* Clean up supported features

molobrakos committed:
eliqonline lib upgrade (#6021)

titilambert committed:
[WIP] Add Fido sensor (#5997)

* Add Fido sensor

* Fix PR #5997 comments

* Make error message usable

lindsaymarkward commented:
@andrey-git - I'm happy to write tests, but I haven't done that before and similar platforms (e.g. yeelight) don't have tests, do they? I added my platform to .coveragerc along with many others that don't have tests. If it's expected, please point me to a best practice example similar enough to mine, thank you!

lindsaymarkward committed:
Bump yeelightsunflower library version (now uses logging)

lindsaymarkward committed:
Remove property unique_id as requested by @pvizeli

titilambert committed:
Update to pyfido 0.1.4 (#6033)

emlove committed:
cmus remove IO from properties (#6030)

balloob committed:
Simplify customize (#6007)

* Simplify customize

* Maintain glob order

* Have glob overrule domain

pvizeli committed:
[Device Tracker] Remove coro (#6014)

* Remove coro and make flow faster.

* fix lint

* address comments

balloob committed:
Update frontend

joncar committed:
LiteJet: Lights should have the option to dim in the UI. (#6031)

* LiteJet: Lights should have the option to dim in the UI.

* Fix lint error.

balloob committed:
Clean up mock_coro (#6037)

titilambert committed:
Add Ebox sensor component (#5998)

* Add Ebox sensor component

* Fix PR #5998 comments

* Fix logger message

titilambert committed:
[WIP] Hydroquebec plugin now use pyhydroquebec lib (#6000)

* Hydroquebec plugin now use pyhydroquebec lib

* Fix logger message

* Fix platform name

pvizeli committed:
Fix name in logging message (#6039)

Add an optional extended description…

jjmontesl committed:
Mediaplayer clementine remote (#5877)

* Added Clementine Music Player Remote component.

* Remove stale stuff

pvizeli committed:
Protect device_tracker scan interval / TTS logging (#6041)

* Protect device_tracker scan interval / TTS logging

* clear pass

andrey-git committed:
Support zwave glob & domain device settings. (#6046)

* Support zwave glob & domain device settings.

* data[DATA_DEVICE_CONFIG].get() now always return non-None

* Update test

* Update tests2

* Lint fix

lindsaymarkward committed:
Bump yeelightsunflower version requirement to 0.0.4

walkerdb committed:
Add aurora sensor (#6003)

* Add aurora sensor

* allow custom forecast threshold for aurora binary sensor

* move AuroraGateway functionality to Aurora data object to conform with HA standards

lindsaymarkward committed:
Bump yeelightsunflower version requirement to 0.0.5, library is now thread safe

lindsaymarkward commented:
Sorry, but I don't understand what's happening with rebase. I followed https://home-assistant.io/developers/development_catching_up/ (`git rebase upstream/dev`) and did a rebase, not merge. Now my PR has many commits and files that are not mine.

Should I do a new PR and then *never* rebase after during future PR processes?

andrey-git commented:
I'm no git expert, but I always do
`git pull upstream dev`
If there are conflict a resolve them and commit.

lindsaymarkward commented:
New PR for this platform: https://github.com/home-assistant/home-assistant/pull/6060
(I won't try and rebase my fork until this is merged)
